From f5edaf1ba5bc2db38ae0c7f595e17aff9d89dcfa Mon Sep 17 00:00:00 2001
From: Daniel P. Berrange <berrange@redhat.com>
Date: Wed, 21 Dec 2016 10:35:17 +0000
Subject: [PATCH 1/1] Remove bogus \o escape in regex

One of the regexes has a bogus \o instead of plain 'o'. Somehow
this magically worked on all versions of python, until 3.6 came
along and complained

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>
---
 generator.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/generator.py b/generator.py
index 730e456..2c3b667 100755
--- a/generator.py
+++ b/generator.py
@@ -1091,7 +1091,7 @@ def is_integral_type (name):
     return not re.search ("^(unsigned)? ?(int|long)$", name) is None
 
 def is_optional_arg(info):
-    return re.search("^\(?\optional\)?", info) is not None
+    return re.search("^\(?optional\)?", info) is not None
 # Functions returning lists which need special rules to check for errors
 # and raise exceptions.
 functions_list_exception_test = {
-- 
1.7.1

